File: project11_oop_mybank.js
"use strict";
class BankAccount {
    constructor(balance) {
        this.balance = balance;
    }
    deposit(amount) {
        if (amount > 0) {
            this.balance += amount;
            return true;
        }
        return false;
    }
    withdraw(amount) {
        if (amount > 0 && amount <= this.balance) {
            this.balance -= amount;
            return true;
        }
        return false;
    }
    getBalance() {
        return this.balance;
    }
}
const account = new BankAccount(1000);
const balanceElement = document.getElementById('balance');
const amountInput = document.getElementById('amount');
const depositButton = document.getElementById('deposit');
const withdrawButton = document.getElementById('withdraw');
balanceElement.textContent = account.getBalance().toString();
depositButton.addEventListener('click', () => {
    const amount = parseFloat(amountInput.value);
    if (account.deposit(amount)) {
        balanceElement.textContent = account.getBalance().toString();
    }
    amountInput.value = '';
});
withdrawButton.addEventListener('click', () => {
    const amount = parseFloat(amountInput.value);
    if (account.withdraw(amount)) {
        balanceElement.textContent = account.getBalance().toString();
    }
    amountInput.value = '';
});

File: project11_oop_mybank.py
class BankAccount:
    def __init__(self, balance):
        self.balance = balance

    def deposit(self, amount):
        if amount > 0:
            self.balance += amount
            return True
        return False

    def withdraw(self, amount):
        if amount > 0 and amount <= self.balance:
            self.balance -= amount
            return True
        return False

account = BankAccount(1000)

while True:
    print(f'Account Balance: ${account.balance}')
    choice = input('Enter D to deposit, W to withdraw, or Q to quit: ').upper()
    
    if choice == 'D':
        amount = float(input('Enter the deposit amount: '))
        if account.deposit(amount):
            print(f'Deposited ${amount}')
        else:
            print('Invalid deposit amount.')
    elif choice == 'W':
        amount = float(input('Enter the withdrawal amount: '))
        if account.withdraw(amount):
            print(f'Withdrawn ${amount}')
        else:
            print('Invalid withdrawal amount.')
    elif choice == 'Q':
        break
    else:
        print('Invalid choice. Try again.')
